src/nbl/asset/interchange/CIESProfileLoader.cpp

```diff
@@ -0,0 +1,352 @@
+    viewParams.subresourceRange.aspectMask = static_cast<IImage::E_ASPECT_FLAGS>(0);
+    viewParams.subresourceRange.levelCount = viewParams.image->getCreationParameters().mipLevels;
+    viewParams.subresourceRange.layerCount = 1u;
+
+    return asset::SAssetBundle (std::move(meta), {ICPUImageView::create(std::move(viewParams))});
+}
+
+static inline core::vectorSIMDf octahdronUVToDir(float u, float v) {
+    core::vectorSIMDf pos = core::vectorSIMDf(2 * (u - 0.5), 2 * (v - 0.5), 0.0);
+    float abs_x = core::abs(pos.x), abs_y = core::abs(pos.y);
+    pos.z = 1.0 - abs_x - abs_y;
+    if (pos.z < 0.0) {
+        pos.x = core::sign(pos.x) * (1.0 - abs_y);
+        pos.y = core::sign(pos.y) * (1.0 - abs_x);
+    }
+
+    return core::normalize(pos);
+}
+
+static inline std::pair<float,float> sphericalDirToAngles(core::vectorSIMDf dir) {
+    float theta = std::acos(dir.z);
+    float phi = std::atan2(dir.y, dir.x);
+    return { theta, phi < 0 ? phi + 2 * core::PI<float>() : phi };
+}
+
+core::smart_refctd_ptr<asset::ICPUImage>
+CIESProfileLoader::createTexture(const CIESProfile& profile, size_t width, size_t height) {
+    asset::ICPUImage::SCreationParams imgInfo;
+    imgInfo.type = asset::ICPUImage::ET_2D;
+    imgInfo.extent.width = width;
+    imgInfo.extent.height = height;
+    imgInfo.extent.depth = 1u;
+    imgInfo.mipLevels = 1u;
+    imgInfo.arrayLayers = 1u;
+    imgInfo.samples = asset::ICPUImage::ESCF_1_BIT;
+    imgInfo.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
+    imgInfo.format = asset::EF_R16_UNORM;
+    auto outImg = asset::ICPUImage::create(std::move(imgInfo));
+
+    asset::ICPUImage::SBufferCopy region;
+    size_t texelBytesz = asset::getTexelOrBlockBytesize(imgInfo.format);
+    size_t bufferRowLength =
+        asset::IImageAssetHandlerBase::calcPitchInBlocks(width, texelBytesz);
+    region.bufferRowLength = bufferRowLength;
+    region.imageExtent = imgInfo.extent;
+    region.imageSubresource.baseArrayLayer = 0u;
+    region.imageSubresource.layerCount = 1u;
+    region.imageSubresource.mipLevel = 0u;
+    region.bufferImageHeight = 0u;
+    region.bufferOffset = 0u;
+    auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(
+        texelBytesz * bufferRowLength * height);
+
+    double maxValue = profile.getMaxValue();
+    double maxValueRecip = 1.0 / maxValue;
+
+    double vertInv = 1.0 / height;
+    double horiInv = 1.0 / width;
+    char* bufferPtr = reinterpret_cast<char*>(buffer->getPointer());
+    for (size_t i = 0; i < height; i++) {
+        for (size_t j = 0; j < width; j++) {
+            auto dir = octahdronUVToDir(j * vertInv, i * horiInv);
+            auto [theta, phi] = sphericalDirToAngles(dir);
+            double I = profile.sample(core::degrees(theta), core::degrees(phi));
+            uint16_t value = static_cast<uint16_t>(
+                std::clamp(I * maxValueRecip * 65535.0, 0.0, 65535.0));
```

Comment:
remember to add +0.5 so that [0,0.5) ->0 and [0.5,1.5) -> 1

very important so you can actually have a Max Value thats reachable

I just fixed a bug resulting from such quanitzation
https://github.com/Devsh-Graphics-Programming/Nabla/commit/9804926db0c7809958c84496be3137848a3642a0